src/oidc/index.ts

```diff
@@ -0,0 +1,67 @@
+import { UserManager, WebStorageStateStore } from 'oidc-client-ts';
+
+type OidcConfiguration = {
+    issuer: string;
+    redirect_uri: string;
+    client_id: string;
+    response_type: string;
+    scope: string;
+    post_logout_redirect_uri: string;
+};
+
+/**
+ * Fetches the OIDC configuration for the given serverUrl.
+ * @returns {Promise<object>} - A promise resolving to the OIDC configuration.
+ * @throws {Error} - If there is a failure while fetching the OIDC configuration.
+ */
+export const fetchOidcConfiguration = async (): Promise<OidcConfiguration> => {
+    const serverUrlLocalStorage = localStorage.getItem('config.server_url') || '';
+    const oidcUrl = `https://${serverUrlLocalStorage}/.well-known/openid-configuration`;
+
+    try {
+        const response = await fetch(oidcUrl);
+        const data = await response.json();
+
+        const endpoints = {
+            authorization_endpoint: data.authorization_endpoint,
+            token_endpoint: data.token_endpoint,
+            userinfo_endpoint: data.userinfo_endpoint,
+            end_session_endpoint: data.end_session_endpoint,
+        };
+
+        localStorage.setItem('config.oidc_endpoints', JSON.stringify(endpoints));
+
+        return data;
+    } catch (error) {
+        console.error('Failed to fetch OIDC configuration:', error);
+        throw error;
+    }
+};
+
+/**
+ * Initiates the authentication flow by redirecting to the authorization endpoint.
+ * @param {string} redirectUri - The URL to redirect to after authentication.
+ * @param {string} post_logout_redirect_uri - The URL to redirect to after logout.
```

Comment:
why one is camelCase the other is snake_case?

Reply:
will change it 🙏